core/src/main/java/org/apache/accumulo/core/iterators/SortedKeyValueIterator.java

```diff
@@ -147,4 +147,13 @@
    *              if not supported.
    */
   SortedKeyValueIterator<K,V> deepCopy(IteratorEnvironment env);
+
+  /**
+   * Closes the Iterator. This should be overridden by the implementing class that has access to <tt>SortedKeyValueIterator source</tt> provided in the
```

Comment:
The javadoc on this seem rather confusing. What does "has access to `SortedKeyValueIterator source`" mean?

Because this is an interface, we should be clear in defining what this method *should* do. We're notoriously bad at that now.

* What kinds of things do Iterators need to close? (by default, nothing)
* What should an implementation do when closing some external resources throws an Exception?